docs/unpackaged.rst

```diff
@@ -1,14 +1,212 @@
+
+Metadata files where a namespace is conditionally applied to components for insertion into different contexts must replace the namespace with a *token*, which CumulusCI replaces with the appropriate value, an empty string, or a default value.
+
+* ``%%%NAMESPACE%%%`` is replaced with the package’s namespace in any context with a namespace (such as a namespaced org or managed org). Otherwise, it remains blank.
+* ``%%%NAMESPACED_ORG%%%`` is replaced with the package’s namespace in a namespaced org *only*, not in a managed installation. Otherwise, it remains blank.
+    .. note:: This token supports use cases where components in one unpackaged metadata bundle refer to components in another, and the dependency bundle acquires a namespace by being deployed into a namespaced org.
+* ``%%%NAMESPACE_OR_C%%%`` is replaced with the package’s namespace in any context with a namespace (such as a namespaced org or managed org). Otherwise, it is replaced with ``c``, the generic namespace used in Lightning components.
+* ``%%%NAMESPACED_ORG_OR_C%%%`` is replaced with the package's namespace in a namespaced org *only*, not in a managed installation. Otherwise, it is replaced with ``c``, the generic namespace used in Lightning components.
+* ``%%%NAMESPACE_DOT%%%`` is replaced with the package’s namespace in any context with a namespace (such as a namespaced org or managed org) followed by a period (``.``) rather than two underscores.
+    .. note:: This token is used to construct references to packaged Record Types.
+
+..
+
+    Example: A portion of metadata from the Nonprofit Success Pack (NPSP) is stored in a subdirectory under ``unpackaged/post``, meaning it's deployed after the application metadata. This metadata updates a Compact Layout on the ``Account`` object, and references packaged metadata from the application as well as from other managed packages. To complete these tasks, this metadata requires the use of namespace tokens to represent the ``npsp`` namespace, letting CumulusCI automatically adapt the metadata to deploy into managed and unmanaged contexts.
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="UTF-8"?>
+    <CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+        <compactLayouts>
+            <fullName>NPSP_Household_Account</fullName>
+            <fields>Name</fields>
+            <fields>npo02__TotalOppAmount__c</fields>
+            <fields>%%%NAMESPACE%%%Number_of_Household_Members__c</fields>
+            <label>NPSP Household Account</label>
+        </compactLayouts>
+    </CustomObject>
+
+..
+
+    Note that only the reference to the NPSP field ``Number_of_Household_Members__c`` is tokenized. (This field is called ``npsp__Number_of_Household_Members__c`` when installed as part of the managed package.) References to NPSP's own managed package dependency, ``npo02``, are not tokenized because this metadata is always namespaced when installed.
+
```

Comment:
Why this specific field? Also, is this detail in parentheses relevant/helpful for a dev to know for this example to make sense? I honestly can't make heads or tails of it.  To be honest, it feels like we’re currently missing one or two details here about why only this field is the one to get tokenized.

Reply:
There's no specific importance to this field, it's just a good example.

But if it's not clear why this field is tokenized, the explanation so far has failed. Should we talk through it? n.b. this are is _notoriously_ difficult to understand for users and even release engineers.

Reply:
Salesforce has a convention of using `namespace__` in front of the field name to indicate which package a particular field belongs to. This particular field is the only one in the compact layout that is a member of the `npsp` package (this is why it's the only one getting the token). We know this only because we were told so here. 

The detail in the parens does help I believe, it's letting us know that in a "managed" context this field will be display as `nspsp__Number_of_Household_Members__c` in an org. This is useful because then I can see how the `%%%NAMESPACE%%%` token is getting replaced with `npsp__` in a "managed" context.